### PR TITLE
Added support for /Collect argument.

### PR DESCRIPTION
**Pull request by harshjain2**
Added support for /Collect argument.

Related to : https://github.com/Microsoft/vstest/issues/309

### CODE REVIEW

File: src/vstest.console/Resources/Resources.resx [hunk 1]
```diff
@@ -646,4 +646,9 @@
   <data name="VstestDiagLogOutputPath" xml:space="preserve">
     <value>Logging Vstest Diagnostics in file: {0}</value>
   </data>
+  <data name="CollectArgumentHelp" xml:space="preserve">
```
Comment (codito) [inline src/vstest.console/Resources/Resources.resx]: Add aka.ms link to vstest documentation.

**Comment by codito** [comment]
@dotnet-bot test this please

File: src/vstest.console/Resources/Resources.resx [hunk 2]
```diff
@@ -646,4 +646,9 @@
...
+    <value>--Collect|/Collect:&lt;DataCollector FriendlyName&gt;
+      Enables data diagnostic adapter(s) in the test run. Default 
+      settings are used if not specified using settings file. More info here : https://aka.ms/fpzskr</value>
```
Comment (smadala) [inline src/vstest.console/Resources/Resources.resx]: `fpzskr`: please add readable word like `vstest-collect`. Created URL `https://aka.ms/vstest-collect`

File: test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+        [TestMethod]
+        public void GetMetadataShouldReturnRunSettingsArgumentProcessorCapabilities()
+        {
+            var processor = new CollectArgumentProcessor();
+            Assert.IsTrue(processor.Metadata.Value is CollectArgumentProcessorCapabilities);
+        }
+
+        [TestMethod]
+        public void GetExecuterShouldReturnRunSettingsArgumentProcessorCapabilities()
```
Comment (smadala) [inline test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs]: Nit: method name.

File: src/vstest.console/Resources/Resources.resx (hunk 2 above)
Comment (harshjain2) [inline src/vstest.console/Resources/Resources.resx]: done.

File: test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs (hunk 3 above)
Comment (harshjain2) [inline test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs]: done.

File: src/vstest.console/Processors/CollectArgumentProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            this.runSettingsManager = runSettingsManager;
+        }
+
+        public void Initialize(string argument)
+        {
+            // 1. Disable all other data collectors. Enable only those data collectors that are explicitely specified by user.
+            // 2. Check if Code Coverage Data Collector is specified in runsettings, if not add it and also set enable to true.
+
+            // if argument is null or doesn't contain any element, don't do anything.
+            if (string.IsNullOrWhiteSpace(argument))
```
Comment (harshjain2) [inline src/vstest.console/Processors/CollectArgumentProcessor.cs]: done.

**Review by smadala — APPROVED** [review]

**Pull request merged**